pkg/security/api/types.go

```diff
@@ -0,0 +1,86 @@
+package api
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+// PodSecurityPolicySubjectReview checks whether a particular user/SA tuple can create the PodSpec.
+type PodSecurityPolicySubjectReview struct {
+	unversioned.TypeMeta
+
+	// Spec defines specification for the PodSecurityPolicySubjectReview.
+	Spec PodSecurityPolicySubjectReviewSpec
+
+	// Status represents the current information/status for the PodSecurityPolicySubjectReview.
+	Status PodSecurityPolicySubjectReviewStatus
+}
+
+// PodSecurityPolicySubjectReviewSpec defines specification for PodSecurityPolicySubjectReview
+type PodSecurityPolicySubjectReviewSpec struct {
+	// PodSpec is the PodSpec to check.
+	PodSpec kapi.PodSpec
+
+	// User is the user you're testing for.
+	// If you specify "User" but not "Group", then is it interpreted as "What if User were not a member of any groups.
+	// If User and Groups are empty, then the check is performed using *only* the ServiceAccountName in the PodSpec.
+	User string
+
+	// Groups is the groups you're testing for.
+	Groups []string
+}
+
+// PodSecurityPolicySubjectReviewStatus contains information/status for PodSecurityPolicySubjectReview.
+type PodSecurityPolicySubjectReviewStatus struct {
+	// AllowedBy is a reference to the rule that allows the PodSpec.  A `nil`, indicates that it was denied.
+	AllowedBy *kapi.ObjectReference
+
+	// Reason is optional.  It indicates why a request was allowed or denied.
+	Reason string
+
+	// PodSpec is the PodSpec after the defaulting is applied.
+	PodSpec kapi.PodSpec
+}
+
+// PodSecurityPolicySelfSubjectReview checks whether this user/SA tuple can create the PodSpec.
+type PodSecurityPolicySelfSubjectReview struct {
+	unversioned.TypeMeta
+
+	// Spec defines specification the PodSecurityPolicySelfSubjectReview.
+	Spec PodSecurityPolicySelfSubjectReviewSpec
+
+	// Status represents the current information/status for the PodSecurityPolicySelfSubjectReview.
+	Status PodSecurityPolicySubjectReviewStatus
+}
+
+// PodSecurityPolicySelfSubjectReviewSpec contains specification for PodSecurityPolicySelfSubjectReview.
+type PodSecurityPolicySelfSubjectReviewSpec struct {
+	// PodSpec is the PodSpec to check.
+	PodSpec kapi.PodSpec
+}
+
+// PodSecurityPolicyReview checks which service accounts (not users, since that would be cluster-wide) can create the `PodSpec` in question.
+type PodSecurityPolicyReview struct {
```

Comment:
For other reviewers: every editor in the namespace should have access to this API.